ingest/dense.py

```diff
@@ -83,12 +83,17 @@ def set_data_array(
         cells_and_expression_vals = dict(
             filter(lambda k_v: k_v[1] > 0, cells_and_expression_vals.items())
         )
-        return self.set_data_array_gene_cell_names(
-            gene_name, linear_data_id, list(cells_and_expression_vals)
-        )
-        return self.set_data_array_gene_expression_values(
-            gene_name, linear_data_id, cells_and_expression_vals.values()
-        )
+        observed_cells = list(cells_and_expression_vals)
+        observed_values = list(cells_and_expression_vals.values())
+
+        # only return significant data (skip if no expression was observed)
+        if (len(observed_values) > 0):
```

Comment:
Conventional Python:

```suggestion
        if len(observed_values) > 0:
```